### PR TITLE
ENH: Add point set registration example

### PR DESCRIPTION
**Pull request by tbirdso**
Porting `itkJensenHavrdaCharvatTsallisPointSetMetricRegistrationTest` script from `ITK` repository as an example of point-to-point v4 registration.

(@thewtex )

### CODE REVIEW

**Comment by thewtex** [comment]
:+1:

**Comment by thewtex** [comment]
@ntustison suggestions welcome

**Comment by tbirdso** [comment]
@thewtex Could you add insight on the linter check failure (style issue)? From what I can tell the error implies a newline is missing to conform to ITK style, but adding a newline did not fix the issue.

**Comment by dzenanz** [comment]
There is a space after int, `int ` should be `int`. Trim trailing space in the editor should do it. If you are using Visual Studio, Edit Menu->Advanced->Format Document should invoke `clang-format` on the file, thus applying all the formatting changes required.

**Comment by tbirdso** [comment]
Perfect, thank you @dzenanz. Very good to know for future issues!

**Comment by tbirdso** [comment]
Pushed the branch again to trigger re-check. Could not reproduce the build-test-python failure [previously seen](https://github.com/InsightSoftwareConsortium/ITKExamples/pull/188/checks?check_run_id=1817169836).

File: src/Registration/Metricsv4/RegisterTwoPointSets/Documentation.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+It: 4 metric value: -0.0956742
+It: 5 metric value: -0.0956774
+It: 6 metric value: -0.0956784
+It: 7 metric value: -0.0956788
+It: 8 metric value: -0.0956789
+It: 9 metric value: -0.0956789
+numberOfIterations: 10
+Moving-source final value: -0.0956789
+Moving-source final position: [0.9999999481711049, 3.6956646020565954e-9,
+							-3.9958144654776e-8, 1.0000000021678688,
```
Comment (dzenanz) [inline src/Registration/Metricsv4/RegisterTwoPointSets/Documentation.rst:36]: It is probably better if tabs are replaced by spaces.

File: Testing/Temporary/CTestCostData.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+---
```
Comment (dzenanz) [inline Testing/Temporary/CTestCostData.txt:1]: This file should not be added

File: Testing/Temporary/LastTest.log [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Start testing: Feb 01 19:46 Eastern Standard Time
```
Comment (dzenanz) [inline Testing/Temporary/LastTest.log:1]: This file should not be added

**Review by dzenanz — COMMENTED** [review]
Mostly looks good.

**Comment by tbirdso** [comment]
Note that `build-test-python` checks are inconsistently failing due to [Issue 178](https://github.com/InsightSoftwareConsortium/ITKExamples/issues/178). Force-pushing the branch to rerun checks without code changes seems to have temporarily resolved the issue.

File: src/Registration/Metricsv4/RegisterTwoPointSets/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.10.2)
+
+project(RegisterTwoPointSets)
+
+find_package(ITK REQUIRED)
+include(${ITK_USE_FILE})
+
+
+add_executable(RegisterTwoPointSets Code.cxx Code.py)
```
Comment (thewtex) [inline src/Registration/Metricsv4/RegisterTwoPointSets/CMakeLists.txt:9]: Here `Code.py` should be removed -- it is not part of the C++ executable.

Comment (tbirdso) [reply src/Registration/Metricsv4/RegisterTwoPointSets/CMakeLists.txt]: Removed. I'll note here that VS unfortunately does not include `Code.py` with the project when it is not included with the executable, which is more a matter of convenience than functionality. Also note that several other ITKExamples add Python files to the executable, perhaps unnecessarily.

**Review by thewtex — APPROVED** [review]
:1234: :+1:

**Pull request merged**